pkg/machinepool/machinepool.go

```diff
@@ -29,10 +32,48 @@ func NewMachinePoolService() MachinePoolService {
 	return &machinePool{}
 }
 
+// ListMachinePools lists all machinepools (or, nodepools if hypershift) in a cluster
+func (m *machinePool) ListMachinePools(r *rosa.Runtime, clusterKey string, cluster *cmv1.Cluster) error {
+	// Load any existing machine pools for this cluster
+	r.Reporter.Debugf("Loading machine pools for cluster '%s'", clusterKey)
+	isHypershift := cluster.Hypershift().Enabled()
+	var err error
+	var machinePools []*cmv1.MachinePool
+	var nodePools []*cmv1.NodePool
+	machinePools, err = r.OCMClient.GetMachinePools(cluster.ID())
+	if isHypershift {
+		nodePools, err = r.OCMClient.GetNodePools(cluster.ID())
+	}
+	if err != nil {
+		return err
+	}
+
+	if output.HasFlag() {
+		if !isHypershift {
+			err = output.Print(machinePools)
+		} else {
+			err = output.Print(nodePools)
+		}
+		return err
+	}
```

Comment:
Sorry one more NP, we can flatten this out, and remove that else statement
```suggestion
	if output.HasFlag() {
		if !isHypershift {
			return output.Print(machinePools)
		}
		return output.Print(nodePools)
	}
```